bricks/next-builder/src/data-providers/chunks/doLintStoryboard.ts

```diff
@@ -279,6 +293,42 @@ export function doLintStoryboard(storyboard: Storyboard): StoryboardError[] {
         }
         break;
       }
+      case "Context": {
+        const { name, onChange } = node.raw as
+          | ContextConf
+          | CustomTemplateState;
+        let isWarned = false;
+        if (Array.isArray(onChange) && onChange.length > 0) {
+          isWarned = onChange.some(
+            (item) =>
+              ACTIONS_BETTER_REPLACED_BY_TRACK.includes(item?.action) ||
+              (item?.target && item?.properties)
+          );
+        } else if (!Array.isArray(onChange) && isObject(onChange)) {
+          isWarned =
+            ACTIONS_BETTER_REPLACED_BY_TRACK.includes(onChange?.action) ||
+            (onChange?.target && onChange?.properties);
+        }
+        isWarned && addMeta(usingWarnedOnChangeInCtxOrState, name, node, path);
+        break;
+      }
+      case "ResolveLifeCycle": {
+        const { useResolves } = node.rawContainer as BrickLifeCycle;
+        if (useResolves && useResolves.length > 0) {
```

Comment:
1. 为啥必须用 useProvider + transform 才能用 context 代替呢？所有 useResolves 都可以用 context 代替吧？
2. 可以直接使用 AST 的数据：

```suggestion
        if (node.resolves?.length) {
```